### PR TITLE
Bump Pulseaudio to 10.0

### PR DESCRIPTION
**Pull request by alarcher**
Solaris Userland update: https://java.net/projects/solaris-userland/sources/gate/revision/7555

### CODE REVIEW

File: components/desktop/pulseaudio/files/pulse_exec_attr [hunk 1]
```diff
@@ -0,0 +1 @@
+Basic Solaris User:solaris:cmd:RO::/usr/bin/pulseaudio:privs=proc_priocntl,file_dac_read
```
Comment (pyhalov) [inline components/desktop/pulseaudio/files/pulse_exec_attr]: Why is it needed? Does pulseaudio drop them after opening devices?

Comment (pyhalov) [inline components/desktop/pulseaudio/files/pulse_exec_attr]: I mean, audio devices are usually owned by Console User

Comment (alarcher) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: Running with ppriv throws missing privileges when auditing in non --system mode.

Comment (pyhalov) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: What files does it try to read? Can we drop privileges after startup? How does it work on Linux?

Comment (alarcher) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: On Linux it uses capabilities to access what it should and raise the priority within the range assigned in the system configuration but beyond its original priority.
On Solaris it is conservative, without the privilege the process can only lower its priority.

Comment (pyhalov) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: This means, it can read my /etc/shadow in any given time... Can we somehow alleviate this issue? Even setuid application looks more sane :)

Comment (alarcher) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: OK I did not answer your question actually.
I see privilege warnings requiring file_dac_read to start pulseaudio in user-mode, this is not related to the priority.
If you do not see it and everything goes fine for you I will just assume that my machine is cursed.

File: components/desktop/pulseaudio/pulseaudio.p5m [hunk 2]
```diff
@@ -24,15 +24,8 @@ license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
...
-link path=etc/bash_completion.d/pacat target=pulseaudio
-link path=etc/bash_completion.d/pacmd target=pulseaudio
-link path=etc/bash_completion.d/pactl target=pulseaudio
-link path=etc/bash_completion.d/padsp target=pulseaudio
-link path=etc/bash_completion.d/paplay target=pulseaudio
-link path=etc/bash_completion.d/parec target=pulseaudio
-link path=etc/bash_completion.d/parecord target=pulseaudio
-link path=etc/bash_completion.d/pasuspender target=pulseaudio
-file path=etc/bash_completion.d/pulseaudio
+file files/pulse_exec_attr path=etc/security/pulseaudio
```
Comment (pyhalov) [inline components/desktop/pulseaudio/pulseaudio.p5m]: /etc/security/exec_attr.d/pulseaudio... And still it doesn't work

Comment (pyhalov) [inline components/desktop/pulseaudio/pulseaudio.p5m]: Another issue is that pulseaudio should be run with pfexec after this. Also it seems proc_prioup is missing.

**Comment by alarcher** [comment]
Let us drop the privilege thing for now.
Ready for merge if you are happy with testing.

File: components/desktop/pulseaudio/patches/08-oss4.patch [hunk 3]
```diff
@@ -1,225 +1,6 @@
---- pulseaudio-6.0/src/modules/oss/module-oss.c.~1~	2015-02-12 17:10:35.000000000 +0300
```
Comment (pyhalov) [inline components/desktop/pulseaudio/patches/08-oss4.patch:1]: Why do you remove source_get_volume ?

Comment (pyhalov) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: Strange, but it seems to be working :)

Comment (pyhalov) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: But I can't control line-out with this pulseaudio version, only master volume level....

Comment (alarcher) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: I have not modified oss4.patch, I do not understand the question?

Comment (pyhalov) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: You have not modified oss4 patch comparing to what? The diff clearly shows changes.

Comment (alarcher) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: Right, I rebased on oi/hipster before working on the component, not sure what happened.

Comment (pyhalov) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: please, add old patch back :)

Comment (alarcher) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: I am already on it but it requires some porting.

Comment (alarcher) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: No I just screwed up with the patch, should be back.

File: components/desktop/pulseaudio/patches/10-default.patch [hunk 4]
```diff
@@ -0,0 +1,12 @@
+--- pulseaudio-10.0/src/daemon/default.pa.in.orig	2017-04-22 18:41:17.928890906 +0200
```
Comment (pyhalov) [inline components/desktop/pulseaudio/patches/10-default.patch:1]: why this change is necessary?

File: components/desktop/pulseaudio/patches/08-oss4.patch (hunk 3 above)
Comment (pyhalov) [reply components/desktop/pulseaudio/patches/08-oss4.patch:1]: Looks fine, I'd like to test it on VM where I tested my last pulseaudio changes

File: components/desktop/pulseaudio/patches/10-default.patch (hunk 4 above)
Comment (pyhalov) [reply components/desktop/pulseaudio/patches/10-default.patch:1]: OK, found that it's necessary

**Comment by pyhalov** [comment]
It seems I'm ready to merge it, but I have one question. 
Have you checked that all dependent packages build with it ?
For me pkg finds the following list:
```
pkg:/audio/sox
pkg:/gnome/media/gnome-media
pkg:/library/audio/gstreamer/plugin/good
pkg:/library/audio/gstreamer1/plugin/good
pkg:/library/audio/libmikmod
pkg:/library/desktop/gtk-vnc
pkg:/library/desktop/mate/libmatemixer
pkg:/library/desktop/ptlib
pkg:/library/desktop/xdg/libcanberra
pkg:/library/qt4
pkg:/library/qt5
pkg:/media/cdrtools
```

**Comment by pyhalov** [comment]
This looks good: https://abi-laboratory.pro/tracker/timeline/pulseaudio/

File: components/desktop/pulseaudio/files/pulse_exec_attr [hunk 5]
```diff
@@ -0,0 +1 @@
+Basic Solaris User:solaris:cmd:RO::/usr/bin/pulseaudio:privs=proc_priocntl,proc_prioup
```
Comment (pyhalov) [inline components/desktop/pulseaudio/files/pulse_exec_attr]: It seems it can be dropped

Comment (alarcher) [reply components/desktop/pulseaudio/files/pulse_exec_attr]: Done.

**Comment by pyhalov** [comment]
tested rebuilding of dependent packages
